run-xmllint-format.py

```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env python3
+"""A wrapper script around xmllint --format, suitable for linting multiple
+files and to use for continuous integration.
+
+This is an alternative API for the xmllint command line.
+It runs over multiple files and directories in parallel.
+A diff output is produced and a sensible exit code is returned.
+
+Based on https://github.com/DoozyX/clang-format-lint-action.
+"""
+
+import argparse
+import codecs
+import difflib
+import fnmatch
+import io
+import errno
+import multiprocessing
+import os
+import signal
+import subprocess
+import sys
+import traceback
+
+from functools import partial
+
+from subprocess import DEVNULL
+
+DEFAULT_EXTENSIONS = "xml"
+DEFAULT_XMLLINT_FORMAT_IGNORE = ".xmllint-format-ignore"
+
+
+class ExitStatus:
+    SUCCESS = 0
+    DIFF = 1
+    TROUBLE = 2
+
+
+def excludes_from_file(ignore_file):
+    excludes = []
+    with io.open(ignore_file, "r", encoding="utf-8") as f:
+        for line in f:
+            if line.startswith("#"):
```

Comment:
I gently suggest adding:
```
line = line.strip()
```
above this line, i.e. before doing anything else with `line`. That way it is easier to ignore truly empty lines. and comments that have an accidental space or two in front of them are still skipped.